### PR TITLE
Publish release rest function

### PR DESCRIPTION
**Pull request by omidontop**
Related: https://github.com/simonsoft/cms-item/pull/32, https://pds-svn-gbg.pdsvision.net/trac/SubversionCMS/changeset/6735/modules

### CODE REVIEW

**Comment by omidontop** [comment]
- [ ] PublishRelease.java is to be moved to cms-publish-rest.

File: cms-publish-rest/src/main/java/se/simonsoft/cms/publish/rest/PublishRelease.java [hunk 1]
```diff
@@ -15,30 +15,31 @@
...
-@JsonIgnoreProperties(ignoreUnknown = true)
 public class PublishRelease {
 
-    private CmsItem item;
+    private CmsItemRepositem item;
     private Map<String, PublishConfig> config;
     private Map<String, PublishProfilingRecipe> profiling;
-    Map<String, Set<String>> translationExecutions;
-    Map<String, Set<String>> releaseExecutions;
+    private Map<String, Set<String>> translationExecutions;
```
Comment (takesson) [inline cms-publish-rest/src/main/java/se/simonsoft/cms/publish/rest/PublishRelease.java:34]: I don't think execution status should be in this class. This class will only be requested once when loading the react form.

Status should be requested on interval from the existing service:  'publish4/release/status'

File: cms-publish-rest/src/test/java/se/simonsoft/cms/publish/rest/PublishResourceTest.java [hunk 2]
```diff
@@ -156,8 +162,13 @@ public void publishResourceTest() throws Exception {
 														htmlHelperMock,
 														storageFactoryMock,
 														getVelocityEngine());
-		
-		String releaseForm = resource.getReleaseForm(itemId);
+
+		PublishRelease publishRelease = resource.getRelease(itemId);
```
Comment (takesson) [inline cms-publish-rest/src/test/java/se/simonsoft/cms/publish/rest/PublishResourceTest.java:166]: Test coverage failed after moving PublishRelease to 'rest' package.

**Review by takesson — CHANGES_REQUESTED** [review]
I think the Jackson stuff looks good.

File: cms-publish-rest/src/main/java/se/simonsoft/cms/publish/rest/PublishRelease.java (hunk 1 above)
Comment (omidontop) [reply cms-publish-rest/src/main/java/se/simonsoft/cms/publish/rest/PublishRelease.java:34]: They are hidden from the serialized JSON output and only exist there to keep it cleaner and to be able to reuse the PublishRelease for both the HTML and the JSON outputs. I can figure something else if you don't like it.

File: cms-publish-rest/src/test/java/se/simonsoft/cms/publish/rest/PublishResourceTest.java (hunk 2 above)
Comment (omidontop) [reply cms-publish-rest/src/test/java/se/simonsoft/cms/publish/rest/PublishResourceTest.java:166]: It's very possible. I will fix the tests and add one that tests the JSON output once the implementation is satisfactory.

File: cms-publish-rest/src/main/java/se/simonsoft/cms/publish/rest/PublishRelease.java (hunk 1 above)
Comment (takesson) [reply cms-publish-rest/src/main/java/se/simonsoft/cms/publish/rest/PublishRelease.java:34]: Ok, I think the HTML output will be simplified to only a title and the "react element"?

Comment (omidontop) [reply cms-publish-rest/src/main/java/se/simonsoft/cms/publish/rest/PublishRelease.java:34]: True, and the navigation bar elements. I shall cleanup and remove what's not used. Any wishes where to place the template?

**Comment by omidontop** [comment]
Alright so I think I'm done here. I'll commit the tests as soon as you had a chance to review this.

File: cms-publish-rest/src/main/java/se/simonsoft/cms/publish/rest/PublishResource.java [hunk 3]
```diff
@@ -104,71 +101,46 @@ public PublishResource(@Named("config:se.simonsoft.cms.hostname") String hostnam
...
 		if (itemProfilingSet != null) {
 			itemProfilings = itemProfilingSet.getMap();
-			logger.debug("ItemId: {} has: {} configured profiles", itemId, itemProfilings.size());
 		}
-		
+
 		Map<String, PublishConfig> configuration = publishConfiguration.getConfigurationFiltered(itemPublish);
-		
-		// Avoid displaying an empty dialog, too complex to handle in Velocity (probably possible though).
-		if (configuration.isEmpty()) {
```
Comment (takesson) [inline cms-publish-rest/src/main/java/se/simonsoft/cms/publish/rest/PublishResource.java:134]: You really want to remove these error conditions? Then you must implement them in React.

File: cms-publish-rest/src/main/java/se/simonsoft/cms/publish/rest/PublishResource.java [hunk 4]
```diff
@@ -104,71 +101,46 @@ public PublishResource(@Named("config:se.simonsoft.cms.hostname") String hostnam
...
-			if (visible == 0) {
-				throw new IllegalStateException("No publications are configured to be visible.");
-			}
+
+		return new PublishRelease((CmsItemRepositem) item, configuration, itemProfilings);
+	}
+
+	@GET
+	@Path("release")
+	@Produces({MediaType.TEXT_HTML, MediaType.APPLICATION_JSON})
```
Comment (takesson) [inline cms-publish-rest/src/main/java/se/simonsoft/cms/publish/rest/PublishResource.java:124]: Should HTML and JSON be the same gettor? The HTML does not need a full PublishRelease object?

Comment (takesson) [reply cms-publish-rest/src/main/java/se/simonsoft/cms/publish/rest/PublishResource.java:124]: Ok, keep it. Required by objectwriter implementation.

**Review by takesson — CHANGES_REQUESTED** [review]
I would like to keep the familiar error messages for now. We can move them to React in the future if it makes sense.

File: cms-publish-rest/src/main/java/se/simonsoft/cms/publish/rest/PublishResource.java (hunk 3 above)
Comment (omidontop) [reply cms-publish-rest/src/main/java/se/simonsoft/cms/publish/rest/PublishResource.java:134]: Brought them back.

**Review by takesson — APPROVED** [review]

**Pull request merged**